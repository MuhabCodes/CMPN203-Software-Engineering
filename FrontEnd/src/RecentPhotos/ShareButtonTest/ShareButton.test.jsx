import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import ShareBtn from '../Share';
import '@testing-library/jest-dom/extend-expect';

it('it renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ShareBtn />, div);
});

it('renders the button correctly', () => {
  const { getByTestId } = render(<ShareBtn />);
  expect(getByTestId('share-test')).toBeTruthy();
});

it('matches a snapshot', () => {
  const tree = renderer.create(<ShareBtn />).toJSON();
  expect(tree).toMatchSnapshot();
});
