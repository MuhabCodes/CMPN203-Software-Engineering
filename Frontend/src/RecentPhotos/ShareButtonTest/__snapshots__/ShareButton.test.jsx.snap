// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<div
  className="share-container"
>
  <button
    className="share-button"
    data-testid="share-test"
    id="share-link-btn"
    onClick={[Function]}
    type="button"
  >
    <img
      alt=""
      className="share-button-img"
      src="https://img.icons8.com/ios/50/000000/forward-arrow.png"
    />
    Share
  </button>
  <div
    className="share-link"
    id="share-area"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="share-actions"
    >
      <button
        className="cancel-share"
        id="share-cancelbtn"
        onClick={[Function]}
        type="button"
      >
        X
      </button>
    </div>
    <textarea
      id="exp-link"
      name="exploreLink"
      readOnly="https://flick.photos/Explore"
      value="https://flick.photos/Explore"
    >
       
    </textarea>
    <button
      id="copy-link-button"
      onClick={[Function]}
      type="button"
    >
      Copy text
    </button>
  </div>
</div>
`;
